src/Shared/PackageManagers/ProjectManagerFactory.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Copyright (c) Microsoft Corporation. Licensed under the MIT License.
+
+namespace Microsoft.CST.OpenSource.PackageManagers
+{
+    using Contracts;
+    using PackageUrl;
+    using System;
+    using System.Collections.Generic;
+    using System.Net.Http;
+
+    public class ProjectManagerFactory
+    {
+
+        private delegate BaseProjectManager? ConstructProjectManager(IHttpClientFactory httpClientFactory, string destinationDirectory = ".", IManagerProvider? managerProvider = null);
+
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly IManagerProvider? _nuGetProvider;
+        
+        /// <summary>
+        /// The dictionary of project managers.
+        /// </summary>
+        private static readonly Dictionary<string, ConstructProjectManager> ProjectManagers =
+            new(StringComparer.InvariantCultureIgnoreCase)
```

Comment:
As we discussed offline, if this field were non-static and initialized in the constructor instead, the lambdas here would be able to directly access the singleton services they need instead of needing to pass them in via the `ConstructProjectManager` delegate.

By doing that, we could drop the two private fields (for `IHttpClientFactory` and `IManagerProvider?`, drop the corresponding two arguments from `ConstructProjectManager`, etc.